src/main/gov/nasa/jpf/psyco/search/SearchEngine.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2015, United States Government, as represented by the 
+ * Administrator of the National Aeronautics and Space Administration.
+ * All rights reserved.
+ *
+ * The PSYCO: A Predicate-based Symbolic Compositional Reasoning environment 
+ * platform is licensed under the Apache License, Version 2.0 (the "License"); you 
+ * may not use this file except in compliance with the License. You may obtain a 
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0. 
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed 
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR 
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the 
+ * specific language governing permissions and limitations under the License.
+ */
+package gov.nasa.jpf.psyco.search;
+
+import gov.nasa.jpf.constraints.api.ConstraintSolver;
+import gov.nasa.jpf.constraints.api.Valuation;
+import gov.nasa.jpf.constraints.api.ValuationEntry;
+import gov.nasa.jpf.jdart.constraints.Path;
+import gov.nasa.jpf.jdart.summaries.MethodSummary;
+import gov.nasa.jpf.jdart.summaries.SummaryStore;
+import gov.nasa.jpf.psyco.PsycoConfig;
+import gov.nasa.jpf.psyco.search.datastructures.searchImage.EnumerativeImage;
+import gov.nasa.jpf.psyco.search.datastructures.searchImage.StateImage;
+import gov.nasa.jpf.psyco.search.datastructures.searchImage.SymbolicImage;
+import gov.nasa.jpf.psyco.search.transitionSystem.EnumerativeTransitionHelper;
+import gov.nasa.jpf.psyco.search.util.HelperMethods;
+import gov.nasa.jpf.psyco.search.transitionSystem.SymbolicTransitionHelper;
+import gov.nasa.jpf.psyco.search.transitionSystem.TransitionHelper;
+import gov.nasa.jpf.psyco.search.transitionSystem.TransitionSystem;
+import gov.nasa.jpf.psyco.util.PsycoProfiler;
+import gov.nasa.jpf.psyco.util.ResultSaver;
+import gov.nasa.jpf.util.JPFLogger;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The search engine is the central hook-up point for search algorithms and extensions.
+ */
+public class SearchEngine {
+
+  private Logger logger;
+  private String folderName = "default";
+  PsycoConfig pconf;
+
+  public SearchEngine(PsycoConfig pconf) {
+    this.pconf = pconf;
+    logger = JPFLogger.getLogger(HelperMethods.getLoggerName());
+    updateFolderName(pconf);
+  }
+
+  public StateImage executeSearch(SummaryStore store,
+          ConstraintSolver solver) {
+    if (pconf.shouldUseEnumerativeSearch()) {
+      PsycoProfiler.reset();
+      executeEnumerativeSearch(store, solver);
+    }
+    if (pconf.shouldUseSymbolicSearch()) {
```

Comment:
No this "if" should not be exclusive. You could execute both searches within one PSYCO run, if desired.